neps/search_spaces/search_space.py

```diff
@@ -90,107 +94,80 @@ def pipeline_space_from_configspace(
     return pipeline_space
 
 
-def pipeline_space_from_yaml(yaml_file_path: str | Path) -> dict[str, Parameter]:
-    """Reads configuration details from a YAML file and constructs a pipeline space
-    dictionary.
-
-    This function extracts parameter configurations from a YAML file, validating and
-    translating them into corresponding parameter objects. The resulting dictionary
-    maps parameter names to their respective configuration objects.
+def pipeline_space_from_yaml(  # noqa: C901, PLR0912
+    config: str | Path | dict,
+) -> dict[str, Parameter]:
+    """Reads configuration details from a YAML file or a dictionary and constructs a
+    pipeline space dictionary.
 
     Args:
-        yaml_file_path (str | Path): Path to the YAML file containing parameter
-        configurations.
+        config (str | Path | dict): Path to the YAML file or a dictionary containing
```

Comment:
Just fyi, you don't need to specify the types in docstrings anymore, it just pulls it straight from the function definition!